extensions/elasticsearch/elasticsearch-7/src/main/java/com/hazelcast/jet/elastic/ElasticSinkBuilder.java

```diff
@@ -0,0 +1,234 @@
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.Serializable;
+
+import static com.hazelcast.jet.impl.util.Util.checkSerializable;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Builder for Elasticsearch Sink
+ *
+ * Requires {@link #clientSupplier(SupplierEx)} and {@link #mapItemFn(FunctionEx)}.
+ *
+ * @param <T>
+ *
+ * @since 4.1
+ */
+public class ElasticSinkBuilder<T> implements Serializable {
+
+    private String name = "elastic";
+    private SupplierEx<? extends RestHighLevelClient> clientSupplier;
+    private ConsumerEx<? super RestHighLevelClient> destroyFn = RestHighLevelClient::close;
+    private SupplierEx<BulkRequest> bulkRequestSupplier = BulkRequest::new;
+    private FunctionEx<? super T, ? extends DocWriteRequest<?>> mapItemFn;
+    private FunctionEx<? super ActionRequest, RequestOptions> optionsFn = (request) -> RequestOptions.DEFAULT;
+    private int preferredLocalParallelism = 2;
+
+    /**
+     * Set the user-friendly source name for this sink
+     *
+     * @param sinkName user-friendly sink name
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> name(@Nonnull String sinkName) {
+        this.name = requireNonNull(sinkName, "sinkName");
+        return this;
+    }
+
+    @Nonnull
+    public String name() {
+        return name;
+    }
+
+    /**
+     * Set the client supplier
+     *
+     * @param clientSupplier supplier for configure Elasticsearch REST client
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> clientSupplier(SupplierEx<? extends RestHighLevelClient> clientSupplier) {
+        checkSerializable(clientSupplier, "clientSupplier");
+        this.clientSupplier = clientSupplier;
+        return this;
+    }
+
+    @Nonnull
+    public SupplierEx<? extends RestHighLevelClient> clientSupplier() {
+        return clientSupplier;
+    }
+
+    /**
+     * Set the destroy function called on completion, defaults to {@link RestHighLevelClient#close()}
+     *
+     * @param destroyFn destroy function
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> destroyFn(ConsumerEx<? super RestHighLevelClient> destroyFn) {
+        checkSerializable(destroyFn, "destroyFn");
+        this.destroyFn = destroyFn;
+        return this;
+    }
+
+    @Nonnull
+    public ConsumerEx<? super RestHighLevelClient> destroyFn() {
+        return destroyFn;
+    }
+
+    /**
+     * Set the bulkRequestSupplier, defaults to new {@link BulkRequest#BulkRequest()}
+     *
+     * @param bulkRequestSupplier supplier for the bulk request
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> bulkRequestSupplier(SupplierEx<BulkRequest> bulkRequestSupplier) {
```

Comment:
no need for this either, another low level concern

Reply:
There are quite few fields on the `BulkRequest`, we would need to expose those parameters somehow. We provide a default, so the user doesn't have to set it (but may if needs to).